### PR TITLE
Fix two memleak where python objets where not free in some cases

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Fix two memleak where python objets where not free in some cases

### CODE REVIEW

File: six/common/builtins/_util.c [hunk 1]
```diff
@@ -142,11 +143,13 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
     if (cmd_raise_on_empty == Py_True)
         raise = 1;
 
+    // Release the GIL so Python can execute other checks while Go run the subprocess
```
Comment (masci) [inline six/common/builtins/_util.c]: s/run/runs

File: six/common/builtins/_util.c [hunk 2]
```diff
@@ -142,11 +143,13 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
...
     PyGILState_Release(gstate);
     PyThreadState *Tstate = PyEval_SaveThread();
 
     cb_get_subprocess_output(subprocess_args, &c_stdout, &c_stderr, &ret_code, &exception);
 
+    // Acquire the GIL not that Go is done
```
Comment (masci) [inline six/common/builtins/_util.c]: s/not/now ?

File: six/common/builtins/datadog_agent.c [hunk 3]
```diff
@@ -166,6 +166,8 @@ PyObject *headers(PyObject *self, PyObject *args, PyObject *kwargs)
     PyObject *headers_dict = from_yaml(data);
     cgo_free(data);
     if (headers_dict == NULL || !PyDict_Check(headers_dict)) {
+        // if headers_dict is not a dict we don't need to old a ref to it
```
Comment (masci) [inline six/common/builtins/datadog_agent.c]: s/old/hold

File: six/common/builtins/tagger.c [hunk 4]
```diff
@@ -34,6 +34,8 @@ PyObject *buildTagsList(char **tags)
         PyObject *pyTag = PyStringFromCString(tags[i]);
         cgo_free(tags[i]);
         PyList_Append(res, pyTag);
+        // PyList_Append increment refcount on pyTag
```
Comment (masci) [inline six/common/builtins/tagger.c]: s/increment/increments or incremented

**Review by masci — CHANGES_REQUESTED** [review]
👨‍🏫

**Review by truthbk — COMMENTED** [review]
Nice fix! I found a remaining leak in one of the error scenarios, I decided to push some code with a minor refactor to make the chance of leaks or potential double-frees a little slimmer going forward. Let me know what you think.

File: six/common/builtins/_util.c [hunk 5]
```diff
@@ -91,84 +92,80 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
...
+        subprocess_args[i] = NULL;
     }
 
-    subprocess_args[subprocess_args_sz] = NULL;
-    int i;
     for (i = 0; i < subprocess_args_sz; i++) {
         char *subprocess_arg = as_string(PyList_GetItem(cmd_args, i));
 
         if (subprocess_arg == NULL) {
             // cleanup
```
Comment (KSerrania) [inline six/common/builtins/_util.c]: Nit: I guess we can also remove the comment since the cleanup isn't done here anymore?

Comment (masci) [reply six/common/builtins/_util.c]:
```suggestion
```

**Review by masci — APPROVED** [review]

**Review by truthbk — APPROVED** [review]
Looks good! Let's see if this fixes the k8s leak.

**Pull request merged**